components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -1239,6 +1276,62 @@ public Response apisApiIdMonetizePost(String apiId, APIMonetizationInfoDTO body,
         return Response.serverError().build();
     }
 
+    /**
+     * Publish API to given external stores.
+     *
+     * @param apiId API Id
+     * @param externalStoreIds  External Store Ids
+     * @param ifMatch   If-match header value
+     * @param messageContext CXF Message Context
+     * @return Response of published external store list
+     */
+    @Override
+    public Response apisApiIdPublishToExternalStoresPost(String apiId, List<String> externalStoreIds, String ifMatch,
+                                                         MessageContext messageContext) {
+        try {
+            String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            APIIdentifier apiIdentifier = APIMappingUtil.getAPIIdentifierFromUUID(apiId, tenantDomain);
+            API api = apiProvider.getAPI(apiIdentifier);
```

Comment:
We can do a single call to get this.

```
            API api = apiProvider.getAPIbyUUID(apiId, tenantDomain);
```

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/6945/commits/e6d05ebb8d0b632ea87d0651379f4a5495964938#diff-3b19b8a2850e1e1f54616622b325adfbL1034